packages/cosmos_ui_components/lib/components/cosmos_bottom_sheet_header.dart

```diff
@@ -19,14 +19,14 @@ class CosmosBottomSheetHeader extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final isExtended = leading != null || actions != null;
     final theme = CosmosTheme.of(context);
     return Column(
       mainAxisSize: MainAxisSize.min,
       children: [
-        SizedBox(height: CosmosTheme.of(context).spacingXXL),
-        if (leading != null || actions != null)
+        if (isExtended)
```

Comment:
The bottom sheet header had issues. You were right about extra spacing when we have leading/actions. I fixed that properly now with all the combinations.